server/services/store/sqlstore/migrate.go

```diff
@@ -74,6 +70,18 @@ func (s *SQLStore) getMigrationConnection() (*sql.DB, error) {
 }
 
 func (s *SQLStore) Migrate() error {
+	if err := s.EnsureSchemaMigrationFormat(); err != nil {
+		return err
+	}
+	defer func() {
```

Comment:
Just want to confirm the order of defers is correct.  The code has defers executed in the following order:
1. Unlock mutex 
2. Morph engine close
3. Database close
4. deleteOldSchemaMigrationTable

Shouldn't the mutex be released last?

Reply:
I don't think we need to wait for database and morph engine closing to release the mutex (2/5), but we do definitely need to run the `deleteOldSchemaMigrationTable` before releasing. Thanks a lot for catching this, I'll put table deletion as part of the migration process. The sequence will be:

1. Run migrations && `deleteOldSchemaMigrationTable`
2. Unlock cluster mutex
3. Morph engine close
4. Migration specific database connection close

Would you make any other change to this order?

Reply:
Fixed on [0c2342b](https://github.com/mattermost/focalboard/pull/3709/commits/0c2342b496d9cb7a58cdb762e8566b3ee3274186)